File: javascript/src/models/eventTypePatch.ts
// this file is @generated

export interface EventTypePatch {
  description?: string;
  archived?: boolean;
  deprecated?: boolean;
  schemas?: any | null;
  featureFlags?: string[] | null;
  /** The event type group's name */
  groupName?: string | null;
}

export const EventTypePatchSerializer = {
  _fromJsonObject(object: any): EventTypePatch {
    return {
      description: object["description"],
      archived: object["archived"],
      deprecated: object["deprecated"],
      schemas: object["schemas"],
      featureFlags: object["featureFlags"],
      groupName: object["groupName"],
    };
  },

  _toJsonObject(self: EventTypePatch): any {
    return {
      description: self.description,
      archived: self.archived,
      deprecated: self.deprecated,
      schemas: self.schemas,
      featureFlags: self.featureFlags,
      groupName: self.groupName,
    };
  },
};

File: kotlin/lib/src/main/kotlin/models/EventTypePatch.kt
// This file is @generated
package com.svix.kotlin.models

import com.svix.kotlin.MaybeUnset
import com.svix.kotlin.MaybeUnsetStringAnyMapSerializer
import kotlinx.serialization.Serializable

@Serializable
data class EventTypePatch(
    val archived: Boolean? = null,
    val deprecated: Boolean? = null,
    val description: String? = null,
    /** Deprecated, use `featureFlags` instead. */
    val featureFlag: MaybeUnset<String> = MaybeUnset.Unset,
    val featureFlags: MaybeUnset<Set<String>> = MaybeUnset.Unset,
    /** The event type group's name */
    val groupName: MaybeUnset<String> = MaybeUnset.Unset,
    @Serializable(with = MaybeUnsetStringAnyMapSerializer::class)
    val schemas: MaybeUnset<Map<String, Any>> = MaybeUnset.Unset,
)
